packages/node_modules/@node-red/editor-client/src/js/nodes.js

```diff
@@ -1899,571 +2136,375 @@ RED.nodes = (function() {
-                    var link;
-                    if (subflow_map[wire.id] && subflow_map[wire.id].id == n.id) {
-                        link = {source:n.in[wire.port], sourcePort:wire.port,target:output};
+
+            subflow.out.forEach(function (output) {
+                output.wires.forEach(function (wire) {
+                    let link;
+                    if (subflowMap[wire.id] && subflowMap[wire.id].id === subflow.id) {
+                        link = { source: subflow.in[wire.port], sourcePort: wire.port, target: output };
                     } else {
-                        link = {source:node_map[wire.id]||subflow_map[wire.id], sourcePort:wire.port,target:output};
+                        link = { source: nodeMap[wire.id] || subflowMap[wire.id], sourcePort: wire.port, target: output };
                     }
                     addLink(link);
-                    new_links.push(link);
+                    newLinks.push(link);
                 });
                 delete output.wires;
             });
-            if (n.status) {
-                n.status.wires.forEach(function(wire) {
-                    var link;
-                    if (subflow_map[wire.id] && subflow_map[wire.id].id == n.id) {
-                        link = {source:n.in[wire.port], sourcePort:wire.port,target:n.status};
-                    } else {
-                        link = {source:node_map[wire.id]||subflow_map[wire.id], sourcePort:wire.port,target:n.status};
-                    }
-                    addLink(link);
-                    new_links.push(link);
-                });
-                delete n.status.wires;
+
+            subflow.status?.wires.forEach(function (wire) {
+                let link;
+                if (subflowMap[wire.id] && subflowMap[wire.id].id === subflow.id) {
+                    link = { source: subflow.in[wire.port], sourcePort: wire.port, target: subflow.status };
+                } else {
+                    link = { source: subflowMap[wire.id] || subflowMap[wire.id], sourcePort: wire.port, target: subflow.status };
+                }
+                addLink(link);
+                newLinks.push(link);
+            });
+
+            if (subflow.status) {
+                delete subflow.status.wires;
             }
         }
+
         // Order the groups to ensure they are outer-most to inner-most
         var groupDepthMap = {};
-        for (i=0;i<new_groups.length;i++) {
-            n = new_groups[i];
-
-            if (n.g && !new_group_set.has(n.g)) {
-                delete n.g;
+        const groupsId = newGroups.map(function (group) { return group.id; });
+        for (const group of newGroups) {
+            if (group.g && !groupsId.includes(group.g)) {
+                delete group.g;
             }
-            if (!n.g) {
-                groupDepthMap[n.id] = 0;
+            if (!group.g) {
+                groupDepthMap[group.id] = 0;
             }
         }
```

Comment:
How does this part (and the loop below) that is supposed to sort the groups work?